lib/puppet.rb

```diff
@@ -20,8 +20,20 @@
 require 'puppet/external/pson/common'
 require 'puppet/external/pson/version'
 require 'puppet/external/pson/pure'
-require 'gettext-setup'
-require 'locale'
+
+# When running within pupeptserver, the gettext-setup gem is not available, so
+# we need to skip initializing i18n functionality and stub out methods normally 
+# supplied by gettext-setup.
+begin
+  require 'gettext-setup'
+  require 'locale'
+  GETTEXT_AVAILABLE = true
```

Comment:
Since this is in the global ruby namespace to avoid the possibility of collision we could put this inside Puppet, i.e., `Puppet::GETTEXT_AVAILABLE`. Since we've already required 'puppet/version' the Puppet module is defined.